src/spyglass/common/common_ripple.py

```diff
@@ -0,0 +1,289 @@
+import datajoint as dj
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from ripple_detection import Karlsson_ripple_detector, Kay_ripple_detector
+from ripple_detection.core import gaussian_smooth, get_envelope
+from spyglass.common import (Electrode, IntervalList,  # noqa
+                             IntervalPositionInfo,
+                             IntervalPositionInfoSelection, LFPBand, Session)
+from spyglass.common.common_nwbfile import AnalysisNwbfile
+from spyglass.common.dj_helper_fn import fetch_nwb
+
+schema = dj.schema('common_ripple')
+
+RIPPLE_DETECTION_ALGORITHMS = {
+    'Kay_ripple_detector': Kay_ripple_detector,
+    'Karlsson_ripple_detector': Karlsson_ripple_detector,
+}
+
+
+def interpolate_to_new_time(df, new_time, upsampling_interpolation_method='linear'):
+    old_time = df.index
+    new_index = pd.Index(np.unique(np.concatenate(
+        (old_time, new_time))), name='time')
+    return (df
+            .reindex(index=new_index)
+            .interpolate(method=upsampling_interpolation_method)
+            .reindex(index=new_time))
+
+
+@schema
+class RippleLFPSelection(dj.Manual):
+    definition = """
+     -> Session
+     brain_region = 'Hippocampus' : varchar(80)
```

Comment:
why is the brain region part of the primary key? and shouldn't that inherit from `BrainRegion` table?

Reply:
I will admit that it is weird, but I want the primary key to be unique to a session and a grouping of electrodes. I was unsure if it would be fussy to have this be pulled in from the brain region table.